base/CMakeLists.txt

```diff
@@ -98,24 +98,27 @@ target_include_directories(${target} PUBLIC
   $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
 )
 
-target_link_libraries(${target} PUBLIC
+target_link_libraries(${target}
+  PUBLIC
   FairRoot::Alignment
   FairRoot::Tools
   FairRoot::ParBase
   FairRoot::GeoBase
 
   ROOT::Core
-  ROOT::EG
   ROOT::Geom
   ROOT::GeomPainter # TGeoTrack
   ROOT::Hist
   ROOT::MathCore
   ROOT::Matrix
-  ROOT::Net # TSocket
   ROOT::Physics
   ROOT::RIO
   ROOT::Tree
   ${VMCLIB}
+
+  PRIVATE
+  ROOT::EG
+  ROOT::Proof
```

Comment:
Are you sure about this unconditional ROOT::Proof?

We're trying to deprecate it. And I think, we have some `BUILD_PROOF_SUPPORT` option in CMake for this.

Reply:
I am not sure. I only saw that we use a Proof header file. That's why I have added it. If Proof is build only conditional than the dependency should be also conditional for sure.

Reply:
@ChristianTackeGSI,

would this be the right syntax?

`$<$<BOOL:${BUILD_PROOF_SUPPORT>:ROOT::Proof>`

Reply:
I think, that should work.

Alternatively, this is very explicit, but should work as well:
```cmake
if(BUILD_PROOF_SUPPORT)
  target_link_libraries(${target} PRIVATE ROOT::Proof)
endif()
```

Looking again, I think, this is _already_ in `base/CMakeLists.txt` at around line 130?

https://github.com/FairRootGroup/FairRoot/blob/a0ba5f20373308e895614f78b6464e71af5516f4/base/CMakeLists.txt#L128-L131

You might need to change the PUBLIC over to PRIVATE there?

Reply:
@ChristianTackeGSI,

thanks for pointing out the existing code block. I completely overlooked it.
The issue should be fixed with the latest commit.